runtime/test/recipe-test.js

```diff
@@ -123,4 +124,41 @@ describe('recipe', function() {
     assert.lengthOf(recipe.particles, 1);
     assert.lengthOf(recipe.handles, 1);
   });
+
+  const getFirstRecipeHash = async manifestContent => {
+    let loader = new Loader();
+    let manifest = await Manifest.parse(manifestContent,
+        {loader, fileName: './manifest.manifest'});
+    let [recipe] = manifest.recipes;
+    assert.isTrue(recipe.normalize());
+    return recipe.digest();
+  };
+
+  it('generates the same hash regardless of session', async () => {
+    // The `shape` definition below will have an id that includes the manifest
+    // id. We want to make sure that doesn't propagate into causing variance for
+    // the recipe digest.
+    const manifestContent = `
```

Comment:
This new test case is reduced from `Products.recipes`, as that's one obvious case that failed, but any recipe incorporating a handle id that includes the manifest id would demonstrate the issue.